### PR TITLE
Java 11 support

### PR DESCRIPTION
**Pull request by lbudai**
related to #3432 

This PR is just a rebased version of #2047 by @Kokan and @MrAnno .

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/java/CMakeLists.txt [hunk 1]
```diff
@@ -51,12 +48,18 @@ bison_target(ModJavaGrammar
 
 find_package(Java 1.7)
 
-module_switch(ENABLE_JAVA "Enable java module" JAVA_FOUND Java_JAVAH_EXECUTABLE)
+if (JAVA_FOUND)
```
Comment (furiel) [inline modules/java/CMakeLists.txt:51]: This is probably a rebase error. The original `module_switch` should work.

Comment (lbudai) [reply modules/java/CMakeLists.txt]: thanks. resolved

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/java/src/main/java/org/syslog_ng/SyslogNgClassLoader.java [hunk 2]
```diff
@@ -146,18 +146,27 @@ private static int javaVersion() {
     return Integer.parseInt(versionStr[0]);
   }
 
+  private void expandClassPath_java9(URL[] urls) throws Exception {
+    InternalMessageSender.debug("Extend classpath according to java version 9");
+    classLoader = URLClassLoader.newInstance(urls ,classLoader);
```
Comment (furiel) [inline modules/java/src/main/java/org/syslog_ng/SyslogNgClassLoader.java:151]: The space and comma is swapped: `urls ,classLoader` vs `urls, classLoader`.

**Review by furiel — APPROVED** [review]

File: modules/java/src/main/java/org/syslog_ng/SyslogNgClassLoader.java (hunk 2 above)
Comment (lbudai) [reply modules/java/src/main/java/org/syslog_ng/SyslogNgClassLoader.java]: fixed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by furiel — APPROVED** [review]

**Pull request closed without merging**